test/colony.js

```diff
@@ -1720,4 +1721,88 @@ contract("Colony", addresses => {
+    it("should be able to add and remove recovery roles", async () => {
+      const owner = addresses[0];
+      let numRecoveryRoles;
+
+      numRecoveryRoles = await colony.numRecoveryRoles();
+      assert.equal(numRecoveryRoles.toNumber(), 0);
+
+      await colony.setRecoveryRole(addresses[1], { from: owner });
+      await colony.setRecoveryRole(addresses[2], { from: owner });
+
+      numRecoveryRoles = await colony.numRecoveryRoles();
+      assert.equal(numRecoveryRoles.toNumber(), 2);
+
+      await colony.removeRecoveryRole(addresses[2], { from: owner });
+      numRecoveryRoles = await colony.numRecoveryRoles();
+      assert.equal(numRecoveryRoles.toNumber(), 1);
+
+      // Can't remove twice
+      await colony.removeRecoveryRole(addresses[2], { from: owner });
+      numRecoveryRoles = await colony.numRecoveryRoles();
+      assert.equal(numRecoveryRoles.toNumber(), 1);
+    });
+
+    it("should not exit recovery mode without approvals", async () => {
+      const owner = addresses[0];
+      await colony.setRecoveryRole(addresses[1], { from: owner });
+      await colony.setRecoveryRole(addresses[2], { from: owner });
+
+      await colony.enterRecoveryMode({ from: owner });
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      await checkErrorRevert(colony.exitRecoveryMode());
+
+      // 1/3 approve
+      await colony.approveExitRecovery({ from: owner });
+      await checkErrorRevert(colony.exitRecoveryMode());
+    });
+
+    it("should exit recovery mode with sufficient approvals", async () => {
+      const owner = addresses[0];
+      await colony.setRecoveryRole(addresses[1], { from: owner });
+      await colony.setRecoveryRole(addresses[2], { from: owner });
+
+      await colony.enterRecoveryMode({ from: owner });
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      // 2/3 approve
+      await colony.approveExitRecovery({ from: owner });
+      await colony.approveExitRecovery({ from: addresses[1] });
+      await colony.exitRecoveryMode();
+    });
+
+    it("users cannot approve twice", async () => {
+      const owner = addresses[0];
+
+      await colony.enterRecoveryMode({ from: owner });
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      await colony.approveExitRecovery({ from: owner });
+      await checkErrorRevert(colony.approveExitRecovery({ from: owner }));
+    });
+
+    it("should allow editing of general variables", async () => {
+      const owner = addresses[0];
+
+      await colony.enterRecoveryMode({ from: owner });
+
+      const unprotected0 = await web3GetStorageAt(colony.address, 5);
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      const unprotected1 = await web3GetStorageAt(colony.address, 5);
+      assert.notEqual(unprotected0.toString(), unprotected1.toString());
```

Comment:
Should check that it's actually been set correctly!

Reply:
Reasonable.